govcd/media_test.go

```diff
@@ -112,37 +121,150 @@ func (vcd *TestVCD) Test_UploadMediaImage_error_withSameItem(check *C) {
 
 // Tests System function Delete by creating media item and
 // deleting it after.
-func (vcd *TestVCD) Test_DeleteMediaImage(check *C) {
+func (vcd *TestVCD) Test_DeleteMedia(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
 	skipWhenMediaPathMissing(vcd, check)
-	itemName := TestUploadMedia + "5"
+	if vcd.config.VCD.Org == "" {
+		check.Skip("Test_DeleteMedia: Org name not given")
+		return
+	}
+	if vcd.config.VCD.Catalog.Name == "" {
+		check.Skip("Test_DeleteMedia: Catalog name not given")
+		return
+	}
+	org, err := vcd.client.GetOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	check.Assert(org, NotNil)
 
-	uploadTask, err := vcd.vdc.UploadMediaImage(itemName, "upload from test", vcd.config.Media.MediaPath, 1024)
+	catalog, err := org.GetCatalogByName(vcd.config.VCD.Catalog.Name, false)
+	check.Assert(err, IsNil)
+	check.Assert(catalog, NotNil)
+
+	itemName := "TestDeleteMedia"
+	uploadTask, err := catalog.UploadMediaImage(itemName, "upload from test", vcd.config.Media.MediaPath, 1024)
 	check.Assert(err, IsNil)
 	err = uploadTask.WaitTaskCompletion()
 	check.Assert(err, IsNil)
 
-	AddToCleanupList(itemName, "mediaImage", vcd.org.Org.Name+"|"+vcd.vdc.Vdc.Name, "Test_UploadMediaImage")
+	AddToCleanupList(itemName, "mediaCatalogImage", vcd.org.Org.Name+"|"+vcd.config.VCD.Catalog.Name, "Test_DeleteMediaImage")
 
-	mediaItem, err := vcd.vdc.FindMediaImage(itemName)
+	media, err := catalog.GetMediaByName(itemName, true)
 	check.Assert(err, IsNil)
-	check.Assert(mediaItem, Not(Equals), MediaItem{})
+	check.Assert(media, Not(Equals), Media{})
 
-	task, err := mediaItem.Delete()
+	task, err := media.Delete()
 	check.Assert(err, IsNil)
 	err = task.WaitTaskCompletion()
 	check.Assert(err, IsNil)
 
-	mediaItem, err = vcd.vdc.FindMediaImage(itemName)
+	media, err = catalog.GetMediaByName(itemName, true)
```

Comment:
This occurrence of `media` is never used.

Reply:
udpated